client/elasticClient.go

```diff
@@ -109,9 +123,19 @@ func (ec *elasticClient) DoBulkRequest(buff *bytes.Buffer, index string) error {
 		return err
 	}
 
+	ec.updatePrometheusMetrics(index)
+
 	return elasticBulkRequestResponseHandler(res)
 }
 
+func (ec *elasticClient) updatePrometheusMetrics(index string) {
+	reqSize, resSize, duration := ec.getStats()
```

Comment:
maybe the prometheusHandler should expose an `IsEnabled() bool` function so you can early exit here if a disabled handler is in use